src/front/spv/mod.rs

```diff
@@ -1695,84 +1699,111 @@ impl<I: Iterator<Item = u32>> Parser<I> {
-                        Glo::Acos => Mf::Acos,
-                        Glo::Atan => Mf::Atan,
-                        Glo::Sinh => Mf::Sinh,
-                        Glo::Cosh => Mf::Cosh,
-                        Glo::Tanh => Mf::Tanh,
-                        Glo::Atan2 => Mf::Atan2,
-                        Glo::Pow => Mf::Pow,
-                        Glo::Exp => Mf::Exp,
-                        Glo::Log => Mf::Log,
-                        Glo::Exp2 => Mf::Exp2,
-                        Glo::Log2 => Mf::Log2,
-                        Glo::Sqrt => Mf::Sqrt,
-                        Glo::InverseSqrt => Mf::InverseSqrt,
-                        Glo::Determinant => Mf::Determinant,
-                        Glo::Modf => Mf::Modf,
-                        Glo::FMin | Glo::UMin | Glo::SMin | Glo::NMin => Mf::Min,
-                        Glo::FMax | Glo::UMax | Glo::SMax | Glo::NMax => Mf::Max,
-                        Glo::FClamp | Glo::UClamp | Glo::SClamp | Glo::NClamp => Mf::Clamp,
-                        Glo::FMix => Mf::Mix,
-                        Glo::Step => Mf::Step,
-                        Glo::SmoothStep => Mf::SmoothStep,
-                        Glo::Fma => Mf::Fma,
-                        Glo::Frexp => Mf::Frexp, //TODO: FrexpStruct?
-                        Glo::Ldexp => Mf::Ldexp,
-                        Glo::Length => Mf::Length,
-                        Glo::Distance => Mf::Distance,
-                        Glo::Cross => Mf::Cross,
-                        Glo::Normalize => Mf::Normalize,
-                        Glo::FaceForward => Mf::FaceForward,
-                        Glo::Reflect => Mf::Reflect,
-                        Glo::Refract => Mf::Refract,
-                        _ => return Err(Error::UnsupportedExtInst(inst_id)),
-                    };
 
-                    let arg_count = fun.argument_count();
-                    inst.expect(base_wc + arg_count as u16)?;
-                    let arg = {
-                        let arg_id = self.next()?;
-                        self.lookup_expression.lookup(arg_id)?.handle
-                    };
-                    let arg1 = if arg_count > 1 {
-                        let arg_id = self.next()?;
-                        Some(self.lookup_expression.lookup(arg_id)?.handle)
-                    } else {
-                        None
-                    };
-                    let arg2 = if arg_count > 2 {
-                        let arg_id = self.next()?;
-                        Some(self.lookup_expression.lookup(arg_id)?.handle)
+                    if gl_op == Glo::Radians || gl_op == Glo::Degrees {
+                        inst.expect(base_wc + 1)?;
+                        let arg = {
+                            let arg_id = self.next()?;
+                            self.lookup_expression.lookup(arg_id)?.handle
+                        };
+
+                        let pi_div_180 = self.maths_constant_expressions[0];
```

Comment:
oh, my apologies. I didn't realize that baking the constant to `self.math_constant_expressions` would not make this code simpler.

Since we'll have to have this complication anyway, we might as well create this constant on demand.
Would you want to try doing this? I imagine we'd do something like `module.constants.fetch_or_append(...)`.

Reply:
Yeah so the problem I get with doing this is that I get this
```
Function [1] 'main' is invalid:
        Expression [12] can't be introduced - it's already in scope
```
error and I'm not quite sure what to do about it.

Reply:
Perhaps we need to insert the constant expression into `lookup_expression`? But I don't know what key to use for that.

Reply:
Oh - I'm starting to see the problem. Because we're adding an extra expression before the multiply, `result_id` no longer matches the expression handle. The best way to fix this might be to go back to what we were doing before.